src/script/components/InputBar/InputBar.tsx

```diff
@@ -565,13 +566,33 @@ const InputBar = ({
 
   const onGifClick = () => openGiphy(inputValue);
 
+  const ping = () => {
+    void messageRepository.sendPing(conversationEntity).then(() => {
+      window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
+    });
+  };
+
+  const totalConversationUsers = conversationEntity.participating_user_ets().length;
+
   const onPingClick = () => {
     if (conversationEntity && !pingDisabled) {
-      setIsPingDisabled(true);
-
-      messageRepository.sendPing(conversationEntity).then(() => {
-        window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
-      });
+      if (conversationEntity.is1to1() || totalConversationUsers < 3) {
+        setIsPingDisabled(true);
+        ping();
+      } else {
+        PrimaryModal.show(PrimaryModal.type.CONFIRM, {
+          primaryAction: {
+            action: () => {
+              setIsPingDisabled(true);
+              ping();
+            },
```

Comment:
After `setIsPingDisabled(true);` is moved to ping, we can simplify this part to just:
```ts
{
  action: ping,
}
```